### ISSUE TITLE
Inconsistent accuracy - low angular errors and high translation errors for a new dataset

### ISSUE THREAD

**Issue by debaditya-unimelb** [vislearn/LessMore#24]
Dear Eric,

Thank you for sharing your code, and it is really useful. However, when I trained on another dataset without the 3D model, the angular errors are really small (~0.5 degrees mean) whereas the translational errors are large (~0.65m). The dataset contained fewer images (~500 training and ~500 testing) of an indoor environment. The training loss for train object was ~0.005 (@300000 itr), the reprojection training loss was ~5 (@300000 itr), and the end-to-end training loss ~10 and ~6 (@50000 itr). I tried on other datasets having more images (~2500) still the error trend remains the same. So, my question: Do I need to change any of the hyperparameters (apart from the camera matrix?) or this is expected behaviour (low angular errors and high translational errors)?

Also, does the constant depth prior significantly affects the indoor accuracy? I used it as 3 (default), however, the dataset contains large rooms and corridors. 

By the way, is the code of the DSAC* (pytorch) online?

Hoping to hear from you soon.

Regards,
Deb

**Comment by ebrach** [comment]
Hi Deb, I'm travelling at the moment and can only comment quickly. The error behavior you report is somewhat expected.  The larger the distance is between the camera and the scene, you will observe large positional errors even if the rotation is just a bit off. Imagine your scene is kilometres away. If you step one meter to the side, the observed image will not have changed at all (small parallax). 

The depth prior for indoor usually affects the stability of the method more than the accuracy. E.g. a unsuitable depth prior can lead to the network collapsing and you would get no reasonable results at all. If you get good results, than leave the prior as it is. If not, play around with the prior.

I will push to finish the DSAC* code when I'm back, but this is only in August. So expect it online end of August.

Best,
Eric

**Comment by debaditya-unimelb** [comment]
Dear Eric, 
That was helpful, much appreciated. Looking forward to seeing your code. Travel safe.
Regards,
Deb
